climada/entity/exposures/base.py

```diff
@@ -1132,9 +1133,11 @@ def write_hdf5(self, file_name):
         LOGGER.info("Writing %s", file_name)
         store = pd.HDFStore(file_name, mode="w")
         pandas_df = pd.DataFrame(self.gdf)
+        wkb_columns = []
         for col in pandas_df.columns:
             if str(pandas_df[col].dtype) == "geometry":
-                pandas_df[col] = np.asarray(self.gdf[col])
+                pandas_df[col] = gpd.GeoSeries(pandas_df[col]).to_wkb()
+                wkb_columns.append(col)
```

Comment:
We can simplify this with `GeoDataFrame.to_wkb()`:
```python
wkb_columns = self.gdf.columns[self.gdf.dtypes == "geometry"]
pandas_df = self.gdf.to_wkb()  # Done!
```
See https://github.com/geopandas/geopandas/blob/7255c57371eafbff5a09f881db2b905d1f9b28a7/geopandas/geodataframe.py#L1254
(I cannot make this a suggestion on GitHub because there is a deleted line 🤷 )

Reply:
Hm. I thought I have checked that and found that `gdf.to_wkb()` only converts the 'geometry' column itself, leaving all other columns of type "geometry" untouched. Wrong?

Reply:
Code and docs say it converts all geometry-type columns. The code I linked above is equivalent to our original one except for the explicit `str` cast on the returned `dtype`, which apparently is not necessary.